### ISSUE TITLE
segmentation fault (core dumped) error

### ISSUE THREAD

**Issue by arifsha** [xmrig/xmrig#2446]
![aws error](https://user-images.githubusercontent.com/28896109/122400798-0ac2cc00-cf4a-11eb-97b0-1d225398e222.PNG)

PLS help i am testing things in my aws account and getting these errors

can any one help me pls !!

**Comment by Lonnegan** [comment]
The memory of your VM is 93% full. xmrig needs at least 2 GB further RAM for the RandomX file.

**Comment by arifsha** [comment]
> The memory of your VM is 93% full. xmrig needs at least 2 GB further RAM for the RandomX file.

its the same error with 32 gb ram vms  i just  posted this image for example even 64 gb ram vms gives same error ;(

**Comment by markuman** [comment]
I **guess** aws has changed something on their nitro hypervisor to detect and kill miners....

**Comment by Lonnegan** [comment]
Try to rename the program and/or try a different algo than RandomX for testing, e.g. cn-heavy/xhv.

**Comment by markuman** [comment]
yes, on none nitro vms (c4 instances) it still works. on nitro vms (c5 or c5a) it segfaults

**Comment by markuman** [comment]
> Try to rename the program and/or try a different algo than RandomX for testing, e.g. cn-heavy/xhv.

results both also in segfaults

**Comment by markuman** [comment]
and it seems that not all regions are affected.

**Comment by SChernykh** [comment]
It looks like it crashes when it tries to read DMI information (memory modules). Try to build with cmake flag `-DWITH_DMI=OFF`.

**Comment by arifsha** [comment]
thank you all guys for reply 

> It looks like it crashes when it tries to read DMI information (memory modules). Try to build with cmake flag `-DWITH_DMI=OFF`.

ok i will try that   thank you .. :)

**Comment by markuman** [comment]
> It looks like it crashes when it tries to read DMI information (memory modules). Try to build with cmake flag -DWITH_DMI=OFF.

works here

**Comment by arifsha** [comment]
> It looks like it crashes when it tries to read DMI information (memory modules). Try to build with cmake flag `-DWITH_DMI=OFF`.

it worked  thank you so much ....

**Comment by desmont28** [comment]
> > It looks like it crashes when it tries to read DMI information (memory modules). Try to build with cmake flag `-DWITH_DMI=OFF`.
> 
> it worked thank you so much ....

how to make it with cmake flag -DWITH_DMI=OFF. Can someone help me ?

**Comment by hiseller** [comment]
I got same issue 
how to make it with cmake flag -DWITH_DMI=OFF. 
Can someone help me ?

**Comment by desmont28** [comment]
> I got same issue
> how to make it with cmake flag -DWITH_DMI=OFF.
> Can someone help me ?

i already know how to make it with cmake flag -DWITH_DMI=OFF

**Comment by hiseller** [comment]
> 
> 
> > I got same issue
> > how to make it with cmake flag -DWITH_DMI=OFF.
> > Can someone help me ?
> 
> i already know how to make it with cmake flag -DWITH_DMI=OFF

could you please give me help?

**Comment by desmont28** [comment]
> > > I got same issue
> > > how to make it with cmake flag -DWITH_DMI=OFF.
> > > Can someone help me ?
> > 
> > 
> > i already know how to make it with cmake flag -DWITH_DMI=OFF
> 
> could you please give me help?

after you download the xmrig github file, first change CMakeLists.txt line 30 which previously said ON to OFF. after doing that save and then proceed to the next setup...

**Comment by hiseller** [comment]
> fter you download the xmrig github file, first change CMakeLists.txt line 30 which previously said ON to OFF. after doing that save and then proceed to the next setup...

my system ubuntu 18.04, how to edit xmrig?

**Comment by markuman** [comment]
Take a look at https://xmrig.com/docs/miner/build/ubuntu

Instead of `cmake ..`  you simple do `cmake -DWITH_DMI=OFF ..`

**Comment by desmont28** [comment]
> > fter you download the xmrig github file, first change CMakeLists.txt line 30 which previously said ON to OFF. after doing that save and then proceed to the next setup...
> 
> my system ubuntu 18.04, how to edit xmrig?

cd xmrig
nano CMakeLists.txt

**Comment by desmont28** [comment]
> Take a look at https://xmrig.com/docs/miner/build/ubuntu
> 
> Instead of `cmake ..` you simple do `cmake -DWITH_DMI=OFF ..`

@hiseller this can too

**Comment by hiseller** [comment]
> 
> 
> Take a look at https://xmrig.com/docs/miner/build/ubuntu
> 
> Instead of `cmake ..` you simple do `cmake -DWITH_DMI=OFF ..`



> 
> 
> > Take a look at https://xmrig.com/docs/miner/build/ubuntu
> > Instead of `cmake ..` you simple do `cmake -DWITH_DMI=OFF ..`
> 
> @hiseller this can too


Worked as a charm,  thank you very much

**Comment by danuonuo** [comment]
This solution is great!
